nearcache/types.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nearcache
+
+import (
+	"fmt"
+	"math"
+	"time"
+)
+
+// InMemoryFormat specifies how the entry values are stored in the Near Cache.
+type InMemoryFormat int8
+
+const (
+	// InMemoryFormatBinary stores the values after serializing them.
+	InMemoryFormatBinary InMemoryFormat = iota
+	// InMemoryFormatObject stores the values in their original form.
+	InMemoryFormatObject
+)
+
+// EvictionPolicy specifies which entry is evicted.
+type EvictionPolicy int32
+
+const (
+	// EvictionPolicyLRU removes the least recently used entry.
+	EvictionPolicyLRU EvictionPolicy = iota
+	// EvictionPolicyLFU removes the least frequently used entry.
+	EvictionPolicyLFU
+	// EvictionPolicyNone removes no entries.
+	EvictionPolicyNone
+	// EvictionPolicyRandom removes a random entry.
+	EvictionPolicyRandom
+	evictionPolicyCount // internal
+)
+
+// String returns a string representation of the eviction policy.
+func (p EvictionPolicy) String() string {
+	switch p {
+	case EvictionPolicyLRU:
+		return "LRU"
+	case EvictionPolicyLFU:
+		return "LFU"
+	case EvictionPolicyNone:
+		return "NONE"
+	case EvictionPolicyRandom:
+		return "RANDOM"
+	}
+	panic(fmt.Errorf("unknown eviction policy: %d", p))
+}
+
+// Stats contains statistics for a Near Cache instance.
+type Stats struct {
+	// CreationTime is the time the Near Cache was initialized.
+	CreationTime time.Time
+	// LastPersistenceTime is the time of the last completed persistence task.
```

Comment:
How does "when the pre-load feature is enabled" improve the documentation?

Reply:
I think it will make a little bit better so can we add it? I could not understand `PreloaderConfig` is for `persistence` without knowing what you said. At least this way, readers can link `pre-load` and `persistence` task concepts together.

Reply:
Added at: https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/75f6b1fbbe251b134468257bbefade2532e8482a